### ISSUE TITLE
paper-autocomplete orphans md-backdrop DOM fragment when allowClear=true + transitionTo

### ISSUE THREAD

**Issue by bdmac** [miguelcobain/ember-paper#538]
Running against alpha6. I have a `paper-autocomplete` component that specifies an action for `onSelectionChange`. That action ultimately triggers a `transitionTo` another route. Basically autocomplete searches for a model and clicking it should transition to that model's show route automatically.

The event fires and the transition happens and everything **looks** like it's fine until you try to click anything onscreen now and realize that somehow the `paper-backdrop` rendered in the wormhole modal was left in the DOM. Since its z-index is so high nothing is clickable anymore (unless I manually delete the `md-backdrop` in devtools).

Ember inspector shows there is no longer any ember-component attached to the now orphaned `md-backdrop` and there is nothing else in the wormhole target div.

In writing up this report I was able to narrow things down a bit to the presence of `allowClear=true` in my `paper-autocomplete`. If I remove that then no `md-backdrop` is left in the DOM and everything works great.

Specifically this works:
```
{{#paper-autocomplete
      placeholder="Select a Business..."
      search=(perform searchBusinesses)
      searchMessage=false
      onSelectionChange=(action 'selectionChanged') as |business autocomplete|}}
      <span class="item-title">
        <span>{{paper-autocomplete-highlight searchText=autocomplete.searchText label=business.name}} ({{business.subdomain}})</span>
      </span>
    {{/paper-autocomplete}}
```
but this does not:
```
{{#paper-autocomplete
      placeholder="Select a Business..."
      search=(perform searchBusinesses)
      searchMessage=false
      allowClear=true
      onSelectionChange=(action 'selectionChanged') as |business autocomplete|}}
      <span class="item-title">
        <span>{{paper-autocomplete-highlight searchText=autocomplete.searchText label=business.name}} ({{business.subdomain}})</span>
      </span>
    {{/paper-autocomplete}}
```

When `allowClear=true` is specified I am left with the following after the `selectionChanged` event triggers transitionTo:
```
<div id="ember-basic-dropdown-wormhole"><md-backdrop id="ember1658_clone" class="md-click-catcher ember-view md-default-theme ng-leave ng-leave-active"><!----></md-backdrop></div>
```

Note that even with `allowClear=true`, if I do not trigger the `transitionTo` (e.g. I just set some random value in the containing component with mut) then things also work in that I am not left w/the backdrop in the DOM.

**Comment by miguelcobain** [comment]
@xomaczar, do you have any remote idea on what is going on?

/cc @ibarrick

**Comment by ibarrick** [comment]
ember-css-transitions is not correctly destroying the clone for some reason. I'm not sure why it would work without allowClear=true except that it may just be a race condition that only occurs when it has to render that extra DOM.

I can't see anything wrong with `this.transition` in the transition-mixin but that seems to be where the issue is.

**Comment by xomaczar** [comment]
Not really, but I find it interesting that the backdrop element id now has prefix of _clone - I am pretty sure we did not set this explicitly. Need to investigate further...

**Comment by ibarrick** [comment]
@xomaczar It's a clone element created by ember-css-transitions. The problem is most likely that ember-css-transitions sometimes fails to destroy that clone when the user transitions away.

**Comment by xomaczar** [comment]
@ibarrick that makes sense. @miguelcobain are you maintaining ember-css-transitions now?

**Comment by miguelcobain** [comment]
@xomaczar I am.

Looks like something is stopping ember-css-transitions from reaching this point: https://github.com/peec/ember-css-transitions/blob/master/addon/mixins/transition-mixin.js#L175

`didTransitionOut`'s default implementation removes the clone.

**Comment by bdmac** [comment]
Fwiw if it's a race condition it's pretty reliable. In my testing it
happened 100% of the time.

**Comment by xomaczar** [comment]
That's good - let's fix it now.

**Comment by xomaczar** [comment]
@bdmac I've attempted to reproduce your issue but, unfortunately,  I couldn't.
1. What `search=(perform searchBusinesses)` does? Is `perform` a custom helper? What does it do?
2. Please create a sample app/repo where we can reproduce your issue.

**Comment by bdmac** [comment]
It's straight out of the power select docs and us from ember-concurrency.
You guys removed the delay/minLength args and that was how they recommend
doing a debounce basically.

**Comment by xomaczar** [comment]
@bdmac paper-autocomplete@v0.2 with delay/minLength had lots of issues - ember-concurrency is the way to go.

**Comment by bdmac** [comment]
@xomaczar so far no luck creating a sample app that repros this. It happens 100% of the time in our actual app and I've got the sample app as close as I can get it with what would seem relevant to this problem and it's not happening so far.

I've resolved all the other package differences where there is overlap but my sample app IS ember 2.9.1 and my real app is back on 2.4.x still.

**Comment by miguelcobain** [comment]
I think it is related with using transitionTo inside the onSelectionChange action. Are you both including this while trying to reproduce the bug?

Also, which browsers are you using?

**Comment by bdmac** [comment]
Yep I triggered a transition to a second route in the sample app. Using
chrome for both. Will keep trying tomorrow.

**Comment by xomaczar** [comment]
Fixed in #726

**Comment by miguelcobain** [comment]
This should be fixed. Please reopen if issue still persists.
